mlonmcu/context.py

```diff
@@ -287,20 +295,44 @@ def __init__(self, name: str = None, path: str = None, lock: bool = False):
         self.cache = TaskCache()
 
     def create_session(self, label="", config=None):
-        """Create a new session in the current context."""
-        idx = self.session_idx + 1
-        logger.debug("Creating a new session with idx %s", idx)
-        temp_directory = self.environment.paths["temp"].path
-        sessions_directory = temp_directory / "sessions"
-        session_dir = sessions_directory / str(idx)
-        session = Session(idx=idx, label=label, dir=session_dir, config=config)
-        self.sessions.append(session)
-        self.session_idx = idx
-        # TODO: move this to a helper function
-        session_link = sessions_directory / "latest"
-        if os.path.islink(session_link):
-            os.unlink(session_link)
-        os.symlink(session_dir, session_link)
+        if hasattr(self, "latest_session_link_lock"):
```

Comment:
Avoid this, just initialize `self.latest_session_link_lock` with `None`

Reply:
Do you mean this in the `__init__()` ?
```
if latest_session_link_lock:
    self.latest_session_link_lock = filelock.FileLock(
        os.path.join(self.environment.home, ".latest_session_link_lock_lock")
    )
else:
    self.latest_session_link_lock = None
```

and change the condition to 

```
if self.latest_session_link_lock:
```

Reply:
Or the latest_session_link_lock should always be True.
```
self.latest_session_link_lock = filelock.FileLock(
    os.path.join(self.environment.home, ".latest_session_link_lock_lock")
)
```

and the condition check can be skipped

```

```

Reply:
I chose the later implementation.

Reply:
Perfect!